code/script.js

```diff
@@ -218,71 +229,135 @@ const generateBoard = () => {
           <button class="filled-button small" onclick="guess('${person.name}')">Guess</button>
         </div>
       </div>
-    `
-  })
-}
+    `;
+  });
+};
 
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  secret =
+    charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+};
 
 // This function to start (and restart) the game
 const start = () => {
+  onGame ? location.reload() : (onGame = true);
+  score = MAX_ATTEMP;
+  updateScore();
+  resultWrappwer.classList.remove("active");
   // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
+  charactersInPlay = CHARACTERS;
   // What else should happen when we start the game?
-}
-
+  generateBoard(); //attach items in #board
+  setSecret(); //assign the answer in the variable 'secret'
+};
+const updateScore = () => {
+  scoreBoard.innerText = `SCORE : ${score} / ${MAX_ATTEMP}`;
+};
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  const category = questions.options[questions.selectedIndex].parentNode.label;
 
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.value;
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
+  if (!currentQuestion) {
+    //prevent null selection
+    return;
+  }
+  const { category, value } = currentQuestion;
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
+  let isKeep;
+  if (category === "hair" || category === "eyes") {
+    isKeep = secret[category] === value;
+    filterCharacters(isKeep);
+    charactersInPlay = charactersInPlay.filter((person) =>
+      isKeep ? person[category] === value : person[category] !== value
```

Comment:
This is good code! (and what I missed in mine, I didn't write [category] here like you did (found out later) and that resulted in the computer removed the correct person cause it couldnt filter correctly haha. Thanks for helping me out Jin with my messy code!